pkg/globalnet/constants/constants.go

```diff
@@ -28,6 +28,7 @@ const (
 	// The following chains are added as part of GN 2.0 implementation.
 	SmGlobalnetEgressChainForPods            = "SM-GN-EGRESS-PODS"
 	SmGlobalnetEgressChainForHeadlessSvcPods = "SM-GN-EGRESS-HDLS-PODS"
+	SmGlobalnetEgressChainForHeadlessSvcEPs  = "SM-GN-EGRESS-HDLS-EPS"
```

Comment:
I think that we can. However, the name is a bit confusing and it makes rules for different purpose exist in one chain. If there is no side effect(like update issue), it would be clean to add a separate chain.

Reply:
I had a look at the code and thought about it for sometime and I think it should be okay even if we introduce a new chain in between. Can you please verify once in your local setup running this updated Globalnet image on an existing (v0.12.0) Globalnet setup. If you find that there is no issue, we can go with the new chain.

Reply:
@sridhargaddam Confirmed that it worked as expected. Please also see below details:

<details>

- Build the container image for this PR

```bash
git checkout issue/1537
make images
docker tag quay.io/submariner/submariner-globalnet:dev quay.io/submariner/submariner-globalnet:issue1537
```

- Deploy v0.12.0 submariner

```bash
git checkout v0.12.0

make deploy using=external-net,globalnet
```

- Confirm the nat chain for v0.12.0 (`SM-GN-EGRESS-HDLS-EPS` doesn't exist)

```bash
docker exec -it cluster1-worker iptables -t nat -S SUBMARINER-GN-EGRESS
# Warning: iptables-legacy tables present, use iptables-legacy to see them
-N SUBMARINER-GN-EGRESS
-A SUBMARINER-GN-EGRESS -j SUBMARINER-GN-MARK
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-NS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-CLUSTER

docker exec -it cluster2-worker iptables -t nat -S SUBMARINER-GN-EGRESS
# Warning: iptables-legacy tables present, use iptables-legacy to see them
-N SUBMARINER-GN-EGRESS
-A SUBMARINER-GN-EGRESS -j SUBMARINER-GN-MARK
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-NS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-CLUSTER
```

- Test that existing test pass
```bash
make e2e using=external-net,globalnet

• [SLOW TEST:46.771 seconds]
[external-dataplane] Connectivity
/go/src/github.com/submariner-io/submariner/test/external/dataplane/connectivity
.go:41
  should be able to connect from an external app to a pod in a cluster
  /go/src/github.com/submariner-io/submariner/test/external/dataplane/connectivi
ty.go:44
------------------------------

JUnit path was configured: /go/src/github.com/submariner-io/submariner/output/e2
e-junit.xml

JUnit report was created: /go/src/github.com/submariner-io/submariner/output/e2e
-junit.xml

Ran 1 of 1 Specs in 46.822 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestE2E (46.82s)
PASS
```

- Update the globalnet image to the one built from this PR
(submariner-operator doesn't allow directly updating globalnet daemonset because it [reconciles the image](https://github.com/submariner-io/submariner-operator/blob/devel/controllers/submariner/globalnet_resources.go#L78). So, it seems that we need to update the `Submariner` CR).

```bash
kind load docker-image quay.io/submariner/submariner-globalnet:issue1537 --name cluster1
kind load docker-image quay.io/submariner/submariner-globalnet:issue1537 --name cluster2

export KUBECONFIG=output/kubeconfigs/kind-config-cluster1
kubectl get daemonset submariner-globalnet -n submariner-operator -o jsonpath='{range .spec.template.spec.containers[*]}{.name}{" "}{.image}{"\n"}{end}'
submariner-globalnet localhost:5000/submariner-globalnet:local

kubectl get submariner -n submariner-operator -o yaml > /tmp/submariner_cr.txt
kubectl edit submariner -n submariner-operator
kubectl get submariner -n submariner-operator -o yaml > /tmp/submariner_cr_after.txt

diff -urN  /tmp/submariner_cr.txt /tmp/submariner_cr_after.txt | grep -C 2 "imageOverrides"
     debug: false
     globalCIDR: 242.254.1.0/24
+    imageOverrides:
+      submariner-globalnet: quay.io/submariner/submariner-globalnet:issue1537
     namespace: submariner-operator

kubectl get daemonset submariner-globalnet -n submariner-operator -o jsonpath='{range .spec.template.spec.containers[*]}{.name}{" "}{.image}{"\n"}{end}'
submariner-globalnet quay.io/submariner/submariner-globalnet:issue1537
```

```bash
export KUBECONFIG=output/kubeconfigs/kind-config-cluster2
kubectl edit submariner -n submariner-operator

kubectl get daemonset submariner-globalnet -n submariner-operator -o jsonpath='{range .spec.template.spec.containers[*]}{.name}{" "}{.image}{"\n"}{end}'
submariner-globalnet quay.io/submariner/submariner-globalnet:issue1537
```

- Confirm the nat chain for this PR (`SM-GN-EGRESS-HDLS-EPS` exists in the right order)

```bash
docker exec -it cluster1-worker iptables -t nat -S SUBMARINER-GN-EGRESS
# Warning: iptables-legacy tables present, use iptables-legacy to see them
-N SUBMARINER-GN-EGRESS
-A SUBMARINER-GN-EGRESS -j SUBMARINER-GN-MARK
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-EPS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-NS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-CLUSTER

docker exec -it cluster2-worker iptables -t nat -S SUBMARINER-GN-EGRESS
# Warning: iptables-legacy tables present, use iptables-legacy to see them
-N SUBMARINER-GN-EGRESS
-A SUBMARINER-GN-EGRESS -j SUBMARINER-GN-MARK
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-PODS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-HDLS-EPS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-NS
-A SUBMARINER-GN-EGRESS -j SM-GN-EGRESS-CLUSTER
```

- Test that e2e test for this PR succeeds

```bash
git checkout issue/1537

go mod vendor

make "e2e" using="external-net globalnet"


Ran 15 of 28 Specs in 281.317 seconds
SUCCESS! -- 15 Passed | 0 Failed | 7 Pending | 6 Skipped
--- PASS: TestE2E (281.34s)
PASS

```

</details>